### PR TITLE
multilevel development [coarse-constant-representation]

### PR DESCRIPTION
**Pull request by barker29**
Some more features to support multilevel hierarchies.

### CODE REVIEW

File: src/MatrixUtilities.hpp [hunk 1]
```diff
@@ -254,6 +254,8 @@ void Deflate(mfem::DenseMatrix& a, const mfem::Vector& v);
 */
 void orthogonalize_from_constant(mfem::Vector& vec);
 
+void orthogonalize_from_vector(mfem::Vector& vec, const mfem::Vector& wrt);
```
Comment (barker29) [inline src/MatrixUtilities.hpp:253]: I should add a line of documentation.

File: src/MatrixUtilities.hpp [hunk 2]
```diff
@@ -254,6 +254,8 @@ void Deflate(mfem::DenseMatrix& a, const mfem::Vector& v);
 */
 void orthogonalize_from_constant(mfem::Vector& vec);
```
Comment (barker29) [inline src/MatrixUtilities.hpp]: Is this still necessary?

**Review by barker29 — COMMENTED** [review]
I also need to add a test for `lineargraphthree`

File: src/MatrixUtilities.hpp (hunk 2 above)
Comment (chakshinglee) [reply src/MatrixUtilities.hpp]: I guess no. `orthogonalize_from_vector` should be general enough to do what `orthogonalize_from_constant` can do.

File: src/Upscale.hpp [hunk 3]
```diff
@@ -107,39 +107,25 @@ class Upscale : public mfem::Operator
     virtual void OrthogonalizeCoarse(mfem::Vector& vect) const;
     virtual void OrthogonalizeCoarse(mfem::BlockVector& vect) const;
 
-    /// Create a coarse vertex space vector
-    virtual mfem::Vector GetCoarseVector() const;
+    virtual void OrthogonalizeLevel(int level, mfem::Vector& vect) const;
```
Comment (chakshinglee) [inline src/Upscale.hpp]: Like `GetMatrix`, `GetVector`, etc. in this class, can we collapse each of `Orthogonalize`, `Restrict`, `Interpolate`, `BlockOffsets` and possibly `Solve` to only one that takes `level` as an input argument.

**Review by chakshinglee — APPROVED** [review]

**Pull request merged**